textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -153,22 +167,40 @@ public ITuple getNextTuple() throws Exception {
     				return null;
     			}
     		}
-    		
-    		ITuple result = currentTuple;
-    		while (currentTuple != null) {
-    			result = matchTuple(currentDictionaryEntry, currentTuple);
-    			if (result != null) {
-    				advanceCursor();
+    		ITuple result = null;
+	    	while (currentTuple != null) {
+	    		result = matchTuple(currentDictionaryEntry, currentTuple);
+	    		if (result != null) {
+	    			advanceCursor();
+	    			cursor++;
+	    			if (cursor > offset){
+	    				break;
+	    			}
+	    		}
+	    		advanceCursor();

```

Comment:
I am not sure if it is intentional, but I can see that the `advanceCursor();` gets called twice if both `result != null` and `cursor < offset`.